ed25519-dalek/src/hazmat.rs

```diff
@@ -40,14 +43,15 @@ pub struct ExpandedSecretKey {
     // modulo l.
     pub(crate) scalar_bytes: [u8; 32],
```

Comment:
Sidebar: perhaps this should be `canonical_scalar_bytes`?

Reply:
Actually, now that you bring this up again, I'm not sure if this was even the right choice. The original reason I included `scalar_bytes` was because [section 5.1.5](https://www.rfc-editor.org/rfc/rfc8032#section-5.1.5) didn't say you do a modular reduction of the clamped integer when computing the pubkey.[^1] So I kept this by the letter of the law.

But this detail doesn't matter! Yes it is true that sometimes `s.reduce() * H != s * H`, but this is only for `H` that are not in the prime subgroup. The only place `scalar_bytes` is used (outside of defining `scalar`) is in the pubkey generation step `EdwardsPoint::mul_base_clamped(scalar_bytes)`. Since the generator is in the prime order subgroup, I should be able to just do `scalar * ED25519_BASEPOINT`.

Does this make sense? Basically I think we should remove `scalar_bytes`. Lmk and I'll push to this branch

Reply:
#545 already does it as ab alternative to this PR